toolchain/templates/delta.mako

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+<%namespace name="helpers" file="helpers.mako"/>
+
+% if engine == 'batch':
+#SBATCH --nodes=${nodes}
+#SBATCH --ntasks-per-node=${tasks_per_node}
+#SBATCH --cpus-per-task=1
+#SBATCH --job-name="${name}"
+#SBATCH --time=${walltime}
+% if partition:
+#SBATCH --partition=${partition}
+% endif
+% if account:
+#SBATCH --account="${account}"
+% endif
+% if gpu:
+#SBATCH --gpus-per-node=${tasks_per_node}
+% endif
+#SBATCH --output="${name}.out"
+#SBATCH --error="${name}.err"
+#SBATCH --export=ALL
+% if email:
+#SBATCH --mail-user=${email}
+#SBATCH --mail-type="BEGIN, END, FAIL"
+% endif
+% endif
+
+${helpers.template_prologue()}
+
+ok ":) Loading modules:\n"
+cd "${MFC_ROOTDIR}"
+. ./mfc.sh load -c d -m ${'g' if gpu else 'c'}
+cd - > /dev/null
+echo
+
+% for target in targets:
+    ${helpers.run_prologue(target)}
+
+    % if not mpi:
+        ${' '.join([f"'{x}'" for x in profiler ])} "${target.get_install_binpath()}"
+    % else:
+        ${' '.join([f"'{x}'" for x in profiler ])}             \
+            srun \
```

Comment:
For interactive mode, you should maybe explicitly set `-np ${nodes*tasks_per_node}`. This should not interfere with Batch jobs.

Reply:
How does one make this only happen in the interactive case?

Reply:
In all batch files, we currently just add `-np ${nodes*tasks_per_node}`. This should match exactly the number of tasks requested in a `-e batch` job so _should be fine to specify_.

Reply:
Also, it's not really clear to me how interactive mode even works. How does this script know how many nodes and tasks it has?

Reply:
If you needed to conditionally specify `-np` (inline) you could so some python magic like:
```
srun \
  ${f"-np {nodes*tasks_per_node}" if engine == 'interactive' else ''}
```

Reply:
@sbryngelson just to be clear, when we use the `nodes` and `tasks_per_node`, these are the (long) names of the command-line options.

Reply:
See https://github.com/MFlowCode/MFC/blob/43f1865cd98eb1408428a17eec45abb9fdfd7cec/toolchain/mfc/run/run.py#L86-L93

We pass `**ARGS()` where `ARGS()` returns a dictionary of command-line options. `**` turns these key/value pairs into keyword arguments.

Reply:
but what if one does not pass these options?

Reply:
If none are specified, it is as if the user had passed in `-N 1 -n 1` (1 task on one _node_). This makes our expressions easier to handle across batch and interactive. This only creates trouble if running in interactive mode and requesting `-N X -n 1` (where $X > 1$), since it will use $X$ tasks across all nodes it has access to, not (necessarily) spawn one task on each of the $X$ nodes. We could see if specifying `-N` and `-n` to mpirun/srun works everywhere for interactive jobs. Fortunately, this is only a problem when requesting strange configurations in interactive mode. It is not a problem for Batch mode.

Reply:
ah yes. ok, got it. makes sense.